### DISCUSSION TITLE
Cake-Day for "Rotten Scripts" 🎂🎉

### DISCUSSION THREAD

**Discussion by HarshCasper** [HarshCasper/Rotten-Scripts#770] [General]
## Background

Exactly, one year back, I started "Rotten Scripts": It was supposed to be my Scripts dump where I can store all sort of Automation Scripts that I write and use. The original idea behind Rotten-Scripts was inspired by [this](https://github.com/anmol27katyani/JMOC-2019-scripts/) project, that I contributed to during the JIIT Month of Code '2020. Co-incidentally my contributions made me a Top-Performer. I decided to have my dump for Scripts and more.

When I created the repository, I had no specific desire for it to have some external contributions. It was solely going to be my thing, just like many of my other repositories back then. Most of the initial stargazers were from my connections over Github and LinkedIn. 

The fact is acknowledged stars on a repository do not signify the impact of the Project. And to be dead honest, Rotten Scripts, was just a script dump: It was not even supposed to be put in the production. People would have found something useful here, might have not. And the whole point of it was to keep it freely available for everyone. 

After a few months of contribution here, I slowly shifted my focus to working on other projects. Rotten Scripts continued to be my dump, where I can commit some useful scripts that I find or write at my leisure. It was during this time when I found some Open-Source Programs run by Student Developer Communities. And even though applying to them was not on top of the priority, I decided to give it a shot.

And as we would have said accidentally or by luck, Rotten Scripts became the part of a beginner-friendly Open-Source Program named [Student Code-In](https://www.linkedin.com/company/studentcode-in/). 

## Open-Source Programs 

Having "Rotten Scripts" as part of the Student Code-In was no less feat. But at the same time, there were a lot of problems. The Project lacked any appreciable documentation with the Scripts had no information on being used or launched for usage. Maintaining new contributions and bringing onboard new contributors was something, I had never done before. And I had natural apprehensions about the same. 

Luckily the first contributors for Rotten Scripts, made the whole thing easy for me. @seema1711 was the First-Contributor to have raised a Pull Request [here](https://github.com/HarshCasper/Rotten-Scripts/pull/1) and [this](https://github.com/HarshCasper/Rotten-Scripts/pull/7) was the Pull Request merged to the Project.

Many other contributors from Student Code-In started contributing to the Project, which included @mohitacecode @sanket143, @mbcse and @AnkitaxPriya. Their contributions made the Project more stable and diverse in terms of contributions. 

Coming up next, we had [PClub Summer of Code](http://www.pclubsummerofcode.in/), a 2-month Open-Source Program, where I was supposed to help new Contributors to contribute to the Project. As part of the Program, many new contributors were onboarded: @Namyalg @HeroicHitesh @Prahitha @srimani-programmer @Anushka-S10 @Puneet-Pal-Singh 

Their Contributions were phenomenal in stacking up the entire directory structure and having necessary stuff that many Developers look forward to. It was from this initiative, that we started following strict Git principles and guidelines for contributing.

Rotten-Scripts became a part of the [HakinCodes Contributor's Hack](https://hakincodes.tech/ch20.html) and [Semester Long Projects](https://slop.dscdaiict.in/). The diversity of contributions kicked in even more with new contributors like @AkM-2018 @kaustubhgupta @avinashkranjan @AdityaAshvin @kritikaparmar-programmer @vybhav72954. We worked on adding more Scripts, automating stuff with Github Actions and making the Project more seamless for new contributors. 

Rotten-Scripts was a part of many other Open-Source Programs, like [NJACK Winter of Code](https://njackwinterofcode.github.io/) and [KWoC](https://kwoc.kossiitkgp.org/), while some of the Open-Source Programs, that it is a part of, like [Script Winter of Code](http://swoc.tech/), [DevScript Winter of Code](https://devscript.tech/woc), [Cross Winter of Code](https://crosswoc.ieeedtu.in/) and [Mexili Winter of Code](https://mexili.github.io/winter_of_code/#/winter_of_code) continuing. 

With diverse contributions using various languages and libraries, we have developed a community around the project, comprising of scripters and people who love coding to automate the boring stuff. With this thing in mind, the next focus for us would be to make the Project more welcoming to new Contributors and give the Project a better shape. 

## What's next? 

While new contributors and maintainers are being onboarded, the idea of what we would be doing next is too vague. But here are some things that we are boiling up right now: 

- Having a Website for the Project (https://github.com/HarshCasper/Rotten-Scripts/pull/763)
- Re-structuring the Directories and Sub-Directories for easy navigation 
- Making the Github Actions and CI workflow to be more stringent to ensure quality contributions
- Removing the dead-weights from the Project (https://github.com/HarshCasper/Rotten-Scripts/pull/769)
- Make the Code more compliant to modern standards and easy to use 

These are some of the things in plans right now, yet we are open to new ideas and thought-processes that may benefit us in the long run. We are also looking for new maintainers for the Project, and if you feel fit, do consider shooting me a Mail [here](mailto:erbeusgriffincasper@gmail.com). 

## Conclusion

And with this, we conclude this monologue. I appreciate all the hard work and the consistent contributions by all the contributors and maintenance alike, on keeping this Project afloat. Till then, keep contributing and maintaining the spirit of collaboration and contribution. Open Source FTW!

**Comment by iseemasaharan** [comment]
That's really amazing @HarshCasper, I am so lucky that I found you as the mentor, and I learned a lot from you. Thank you so much. And congratulations, this project has helped many students to start with open-source. All the very best, and keep going!!

**Comment by sohamsshah** [comment]
Wow man, this is insane! All I can say that this project is going to the moon 🚀. Open Source can be overwhelming for starters, but by this initiative of yours, it has made Open Source `cake-walk` for them + the mindblowing ideas just make this worth. Congratulations for this and more such amazing achievements await :) 🤩

Also, everyone who were a part of this, be the contributors or fellow mentors; big appreciation and congratulations to you as well! Cheers 😋

And of course, for me this has been an amazing experience here; super exciting and motivating. Have learnt a lot from you, @seema1711 and @vybhav72954 🔥 Thank you :)
